### DISCUSSION TITLE
Using a hook to filter out certain models after execution but before return

### DISCUSSION THREAD

**Discussion by SirensOfTitan** [sqlalchemy/sqlalchemy#8956] [Usage Questions]
What I want to do in a nutshell:

1. Load models from database using a query
2. Before query returns: filtering out models based on some criteria specified on each model class as a static method, ignoring non model based results (e.g. a count).
3. Return only the models that match the criteria.

... I'm a little new with SQLAlchemy, so any help is appreciated!

**Comment by CaselIT** [comment]
Hi,

Usually a sql condition to place in the where is used to filter the loads. You may also use the `SessionEvents.do_orm_execute` event to intercept all queries and attacth a condition to them 

Regarding your question I'm not sure there is a way of using the event system, like`InstanceEvents.load`, to filter the returned models. @zzzeek is something like this supported?

**Reply by zzzeek** [reply]
you could theoretically use `do_orm_execute` to filter results but this would be awkward, as the event would need to decide for which queries it needs to take effect and not for those that it doesn't apply towards.    

that is to say, I wouldn't normally want to do what the poster is suggesting and I would be approaching the problem differently.  that said, sure you could use [do_orm_execute](https://docs.sqlalchemy.org/en/14/orm/events.html#sqlalchemy.orm.SessionEvents.do_orm_execute) to do this in conjunction with [column_descriptions](https://docs.sqlalchemy.org/en/14/orm/queryguide.html#inspecting-entities-and-columns-from-orm-enabled-select-and-dml-statements) so that you know what you're getting.
